aws-common/src/main/java/io/druid/common/aws/AWSEndpointConfig.java

```diff
@@ -0,0 +1,52 @@
+/*
+ * Licensed to Metamarkets Group Inc. (Metamarkets) under one
+ * or more contributor license agreements. See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership. Metamarkets licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License. You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied. See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+package io.druid.common.aws;
+
+import com.fasterxml.jackson.annotation.JsonProperty;
+
+public class AWSEndpointConfig
```

Comment:
Why do we need to have this class instead of relying on https://github.com/aws/aws-sdk-java/blob/1.11.291/aws-java-sdk-core/src/main/java/com/amazonaws/SDKGlobalConfiguration.java ?

Reply:
Sorry, I'm not sure how to use that. Does `SDKGlobalConfiguration` provide an endpoint configuration?

Reply:
They are system properties the sdk picks up as per: https://github.com/aws/aws-sdk-java/blob/4734de6fb0f80fe5768a6587aad3b9d0eaec388f/aws-java-sdk-core/src/main/java/com/amazonaws/auth/SystemPropertiesCredentialsProvider.java#L35

Reply:
Basically if there's functionality in this config that is not available in the SDK supported system properties and env vars, then that is fine. But if all the config stuff can be done in SDK idiomatic ways, then I suggest we simply document where to find info on the SDK.

Reply:
@drcrallen the linked doc is for credentials and we're already using it for credentials.

AFAIK, aws-java-sdk doesn't provide such a configuration chain for the endpoint configuration. Instead, it's recommended to use `AwsClientBuilder.EndpointConfiguration` (https://docs.aws.amazon.com/AWSJavaSDK/latest/javadoc/com/amazonaws/client/builder/AwsClientBuilder.EndpointConfiguration.html) or `AmazonS3Client.setEndpoint()` (https://docs.aws.amazon.com/AWSJavaSDK/latest/javadoc/com/amazonaws/AmazonWebServiceClient.html#setEndpoint-java.lang.String-java.lang.String-java.lang.String-). I used the later one even though it's deprecated because using `AwsClientBuilder` causes some injection problem. I didn't dig that problem and not sure how to fix it.

Reply:
As another example: https://github.com/aws/aws-sdk-java/blob/1.11.293/aws-java-sdk-core/src/main/java/com/amazonaws/regions/LegacyRegionXmlMetadataBuilder.java allows specifying region endpoints. But those might be such an esoteric way of specifying things then it could be better to just handle it on our own until more obvious ways arise.